core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -619,6 +641,21 @@ class KafkaApis(val requestChannel: RequestChannel,
             request.header.clientId,
             topicPartition,
             status.error.exceptionName))
+
+          if (request.header.apiVersion >= 10) {
+            status.currentLeader = {
+              status.error match {
+                case Errors.NOT_LEADER_OR_FOLLOWER =>
+                  val leaderNode = getCurrentLeader(topicPartition, request.context.listenerName)
+                  nodeEndpoints.put(leaderNode.node.id(), leaderNode.node)
+                  new LeaderIdAndEpoch()
+                    .setLeaderId(leaderNode.leaderId)
+                    .setLeaderEpoch(leaderNode.leaderEpoch)
+                case _ =>
+                  null
```

Comment:
could this just be the default leaderIdAndEpoch?

Reply:
yeah, looking again the currentLeader should already be set to the default, I removed the allocation